plugins/modules/proxmox_pool.py

```diff
@@ -0,0 +1,180 @@
+    type: str
+
+extends_documentation_fragment:
+    - community.general.proxmox.documentation
+    - community.general.attributes
+"""
+
+EXAMPLES = """
+- name: Create new Proxmox VE pool
+  community.general.proxmox_pool:
+    api_host: node1
+    api_user: root@pam
+    api_password: password
+    poolid: test
+    comment: 'New pool'
+
+- name: Delete the Proxmox VE pool
+  community.general.proxmox_pool:
+    api_host: node1
+    api_user: root@pam
+    api_password: password
+    poolid: test
+    state: absent
+"""
+
+RETURN = """
+poolid:
+  description: The pool ID.
+  returned: success
+  type: str
+  sample: test
+msg:
+  description: A short message on what the module did.
+  returned: always
+  type: str
+  sample: "Pool test successfully created"
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.proxmox import (proxmox_auth_argument_spec, ProxmoxAnsible)
+
+
+class ProxmoxPoolAnsible(ProxmoxAnsible):
+
+    def is_pool_exists(self, poolid):
+        """Check whether pool already exist
+
+        :param poolid: str - name of the pool
+        :return: bool - is pool exists?
+        """
+        try:
+            pools = self.proxmox_api.pools.get()
+            for pool in pools:
+                if pool['poolid'] == poolid:
+                    return True
+            return False
+        except Exception as e:
+            self.module.fail_json(msg="Unable to retrieve pools: {0}".format(e))
+
+    def is_pool_empty(self, poolid):
+        """Check whether pool has members
+
+        :param poolid: str - name of the pool
+        :return: bool - is pool empty?
+        """
+        return True if not self.get_pool(poolid)['members'] else False
+
+    def create_pool(self, poolid, comment=None):
+        """Create Proxmox VE pool
+
+        :param poolid: str - name of the pool
+        :param comment: str, optional - Description of a pool
+        :return: None
+        """
+        if self.is_pool_exists(poolid):
+            self.module.exit_json(changed=False, poolid=poolid, msg="Pool {0} already exists".format(poolid))
+
+        if self.module.check_mode:
+            self.module.exit_json(changed=True, poolid=poolid, msg="Pool {0} already exists".format(poolid))
+
+        try:
+            self.proxmox_api.pools.post(poolid=poolid, comment=comment)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to create pool with ID {0}: {1}".format(poolid, e))
+
+    def delete_pool(self, poolid):
+        """Delete Proxmox VE pool
+
+        :param poolid: str - name of the pool
+        :return: None
+        """
+        if not self.is_pool_exists(poolid):
+            self.module.exit_json(changed=False, poolid=poolid, msg="Pool {0} doesn't exist".format(poolid))
+
+        if self.module.check_mode:
+            self.module.exit_json(changed=True, poolid=poolid, msg="Pool {0} doesn't exist".format(poolid))
```

Comment:
I would do
```suggestion
            return
```
so that `Pool {0} successfully deleted` is returned as the message.

(If you don't want this, you need to update the message - the one used right now is wrong.)